lib/puppet/provider/foreman_smartproxy/rest_v2.rb

```diff
@@ -2,6 +2,12 @@
 
   confine :feature => :apipie_bindings
 
+  def raise_error(e)
+    body = JSON.parse(e.response)["error"]["full_messages"].join(" ") rescue 'N/A'
+    fail("Error while registering proxy (#{e.message}): #{body}, " +
+      "registration can be skipped by unsetting --register-in-foreman option.")
```

Comment:
This is only relevant for the installer. Can we come up with something that also works in all cases? I'd prefer if all users registered the proxy so I wonder if we should even encourage users to disable registration.